06.Lesson_ThinkingProperties/ThinkingPropertiesExercise/PopsicleFinance/README.md

```diff
@@ -0,0 +1,29 @@
+(Did not have as much time to work on and organize this one)
+
+# Categories
+
+## Unit tests
+- ❌ Unit test for `deposit` - check that rewards and balances are updated
```

Comment:
I am having a very hard time writing this test. It's failing at https://github.com/AbhiGulati/Certora-Tutorials/pull/1/commits/1e9ddafe6c15a45d6bdf8e914359ecb0a27c10ad for reasons I can't understand. It seems like my ghosts are not working as expected. 

The contract sets the user's `feesCollectedPerShare` to the contract-level `totalFeesEarnedPerShare`
```
accounts[msg.sender].feesCollectedPerShare = totalFeesEarnedPerShare;
```
and in my rule I am just trying to assert that these things are equal after `deposit`, but it is always failing. Do you see anything wrong with my rules or ghosts and hooks?

Reply:
I got this working by correctly using an `Sload` hook